Client/Frontend/UserContent/UserScripts/FarblingProtection.js

```diff
@@ -0,0 +1,304 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"use strict";
+
+((params) => {
+  // A value between 0.99 and 1 to fudge the audio data
+  // A value between 0.99 to 1 means the values in the destination will 
+  // always be within the expected range of -1 and 1.
+  // This small decrease should not affect affect legitimite users of this api.
+  // But will affect fingerprinters by introducing a small random change.
+  const fudgeFactor = params['fudgeFactor']
+  // Fake data that is to be used to construct fake plugins
+  const fakePluginData = params['fakePluginData']
+  // A value representing a fake voice name that will be used to add a fake voice
+  const fakeVoiceName = params['fakeVoiceName']
+  // This value is used to get a random index between 0 and voices.length
+  // It's important to have a value between 0 - 1 in order to be within the 
+  // array bounds
+  const randomVoiceIndexScale = params['randomVoiceIndexScale']
+
+  const farbledChannels = new WeakMap()
+  const braveNacl = window.nacl
+  delete window.nacl
+
+  const farbleArrayData = (destination) => {
+    // Let's fudge the data by our fudge factor.
+    for (const index in destination) {
+      destination[index] = destination[index] * fudgeFactor
+    }
+  }
+
+  // Convert an unsinged byte (Uint8) to a hex character
+  // Unsigned bytes must be between 0 and 255
+  const byteToHex = (unsignedByte) => {
+    // convert the possibly signed byte (-128 to 127) to an unsigned byte (0 to 255).
+    // if you know, that you only deal with unsigned bytes (Uint8Array), you can omit this line
+    // const unsignedByte = byte & 0xff
+
+    // If the number can be represented with only 4 bits (0-15),
+    // the hexadecimal representation of this number is only one char (0-9, a-f).
+    if (unsignedByte < 16) {
+      return '0' + unsignedByte.toString(16)
+    } else {
+      return unsignedByte.toString(16)
+    }
+  }
+
+  // Convert an array of unsigned bytes (Uint8Array) to a hex string.
+  // Each value in the array must be between 0 and 255,
+  // resulting in hex values between 0 to f (i.e. 0 to 15)
+  const toHexString = (unsignedBytes) => {
+    return Array.from(unsignedBytes)
+      .map(byte => byteToHex(byte))
+      .join('')
+  }
+
+  // Hash an array
+  const hashArray = (a) => {
+    const byteArray = new Uint8Array(a.buffer)
+    const hexArray = braveNacl.hash(byteArray)
+    return toHexString(hexArray)
+  }
+
+  // 1. Farble `getChannelData`
+  // This will also result in a farbled `copyFromChannel`
+  const getChannelData = window.AudioBuffer.prototype.getChannelData
+  window.AudioBuffer.prototype.getChannelData = function () {
+    const channelData = Reflect.apply(getChannelData, this, arguments)
+    let hashes = farbledChannels.get(channelData)
+
+    // First let's check if we already farbled this set
+    if (hashes !== undefined) {
```

Comment:
If `hashes` is null, it will enter the if-statement here.
You probably just want `if (hashes)`